src/app/pages/applications/catalog/catalog.component.ts

```diff
@@ -0,0 +1,219 @@
+import { Component, OnInit, Output, EventEmitter } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { TranslateService } from '@ngx-translate/core';
+import { Router } from '@angular/router';
+import { Subject, Subscription } from 'rxjs';
+import { CoreService, CoreEvent } from 'app/core/services/core.service';
+
+import { EntityJobComponent } from '../../common/entity/entity-job/entity-job.component';
+import { EntityToolbarComponent } from 'app/pages/common/entity/entity-toolbar/entity-toolbar.component';
+import { EntityUtils } from '../../common/entity/utils';
+import { DialogFormConfiguration } from '../../common/entity/entity-dialog/dialog-form-configuration.interface';
+import { DialogService, SystemGeneralService } from '../../../services/index';
+import { ModalService } from '../../../services/modal.service';
+import { ApplicationsService } from '../applications.service';
+
+import { KubernetesSettingsComponent } from '../forms/kubernetes-settings.component';
+import { ChartReleaseAddComponent } from '../forms/chart-release-add.component';
+import { PlexFormComponent } from '../forms/plex-form.component';
+import { NextCloudFormComponent } from '../forms/nextcloud-form.component';
+import  helptext  from '../../../helptext/apps/apps';
+
+@Component({
+  selector: 'app-catalog',
+  templateUrl: './catalog.component.html',
+  styleUrls: ['../applications.component.scss']
+})
+export class CatalogComponent implements OnInit {
+  @Output() switchTab = new EventEmitter<string>();
+
+  public catalogApps = [];
+  private dialogRef: any;
+  private poolList = [];
+  private selectedPool = '';
+  public settingsEvent: Subject<CoreEvent>;
+  private kubernetesForm: KubernetesSettingsComponent;
+  private chartReleaseForm: ChartReleaseAddComponent;
+  private plexForm: PlexFormComponent;
+  private nextCloudForm: NextCloudFormComponent;
+  private refreshForm: Subscription;
+  private refreshTable: Subscription;
+
+  public choosePool: DialogFormConfiguration = {
+    title: helptext.choosePool.title,
+    fieldConfig: [{
+      type: 'select',
+      name: 'pools',
+      placeholder: helptext.choosePool.placeholder,
+      required: true,
+      options: this.poolList
+    }],
+    method_ws: 'kubernetes.update',
+    saveButtonText: helptext.choosePool.action,
+    customSubmit: this.doPoolSelect,
+    parent: this,
+  }
+
+  constructor(private dialogService: DialogService,
+    private mdDialog: MatDialog, private translate: TranslateService,
+    private router: Router, private core: CoreService, private modalService: ModalService,
+    private appService: ApplicationsService, private sysGeneralService: SystemGeneralService) { }
+
+  ngOnInit(): void {
+    this.appService.getAllCatalogItems().subscribe(res => {
+      if (Object.keys(res[0].trains.test).length > 0) {
```

Comment:
Should we switch to stable trains, @sonicaj  ?

Reply:
We should but can we please do that like at the very end when all the reviewers have approved ? I am building this branch and going to test it out before approving :) So would be easier if it is like this for now.